core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -342,10 +342,15 @@ class LogCleaner(initialConfig: CleanerConfig,
       } catch {
         case e @ (_: ThreadShutdownException | _: ControlThrowable) => throw e
         case e: Exception =>
-          if (currentLog.isEmpty) {
-            throw new IllegalStateException("currentLog cannot be empty on an unexpected exception", e)
+          val erroneousLog = e match {
+            case e: ErroneousLogException =>
+              e.log
+            case _ =>
+              if (currentLog.isEmpty) {
```

Comment:
It seems like the only reason we have `currentLog` is for this catch block. I'm wondering if we can remove it and rely on the `ErroneousLogException` more generally. In other words, rather than setting `currentLog` before calling `cleanLog`, we can just let it raise `ErroneousLogException` in the case of an error. What do you think?

Reply:
Sounds good to me. The new logic polluted the method with try/catch blocks, so I split it into two methods